pxf/pxf-hbase/src/main/java/org/apache/hawq/pxf/plugins/hbase/HBaseFilterBuilder.java

```diff
@@ -89,12 +90,15 @@ private boolean filterNotOpPresent(String filterString) {
      * @throws Exception if parsing failed
      */
     public Filter getFilterObject(String filterString) throws Exception {
+        if (filterString == null)
+            return null;
+
         // First check for NOT, HBase does not support this
         if (filterNotOpPresent(filterString))
             return null;
 
         FilterParser parser = new FilterParser(this);
-        Object result = parser.parse(filterString);
+        Object result = parser.parse(filterString.getBytes());
```

Comment:
this is confusing ... we already have a string here, then we convert it to byte[] to send to parser which converts each byte to char ... why ?